wandb/sdk/launch/agent/agent.py

```diff
@@ -516,7 +518,11 @@ async def run_job(
         Arguments:
             job: Job to run.
         """
-        _msg = f"{LOG_PREFIX}Launch agent received job:\n{pprint.pformat(job)}\n"
+        job_copy = copy.deepcopy(job)
+        if "runSpec" in job_copy and "_wandb_api_key" in job_copy["runSpec"]:
+            job_copy["runSpec"]["_wandb_api_key"] = "<redacted>"
```

Comment:
This looks a little scary---what if someone adds a new entry containing something like an API key in the future? How will they know to update this function?